custom_model_runner/datarobot_drum/runtime_parameters/runtime_parameters.py

```diff
@@ -33,11 +33,44 @@ class RuntimeParameters:
 
     PARAM_PREFIX = "MLOPS_RUNTIME_PARAM"
 
+    # Used to determine if a user has specified a default or not since None is a valid
+    # user input.
+    _UNSET = object()
+
     @classmethod
-    def get(cls, key):
+    def get(cls, key, default=_UNSET):
```

Comment:
since a default is required if the key is missing otherwise error, perhaps two methods:

- `get_or_default(key, default)`
- `strict_get(key)`

the names are first off the top of my head, but you get the point.

Reply:
> since a default is required

do we have to make the default required? I'd keep the dict semantics of `get` + `[]` if possible.

Reply:
This is basically the same behavior as `ConfigParser.get()` in the stdlib: If you ask for an option that isn't found in the config file and you didn't specify a `fallback` option the it will raise `NoOptionError`, otherwise it will return the fallback. https://github.com/python/cpython/blob/3.11/Lib/configparser.py#L781

If we don't want to move forward with this approach, I'd say I'm just going to close this PR out and wait for more user feedback.

Reply:
good point about being consistent with `ConfigParser.get()`.

Reply:
If you want to follow ConfigParser.get, which seems fine, then maybe change the kwarg to `fallback`?  also, update the error.  the issue i see with `get` is that many users will first assume `dict.get` where `None` is the default fallback.  so maybe the method needs a more descriptive name?  :shrug: